core/ProfileDetailGenerator.py

```diff
@@ -8,20 +9,43 @@ def __init__(self, profiles, mapping_type_code, blacklistedValueSets, fields_to_
         self.profiles = profiles
         self.mapping_type_code = mapping_type_code
         self.fields_to_exclude = fields_to_exclude
-        self.simple_data_types = ["instant", "time", "date", "dateTime", "decimal", "boolean", "integer", "string", "uri", "base64Binary", "code", "id", "oid", "unsignedInt", "positiveInt", "markdown", "url", "canonical", "uuid"]
+        self.simple_data_types = ["instant", "time", "date", "dateTime", "decimal", "boolean", "integer", "string",
+                                  "uri", "base64Binary", "code", "id", "oid", "unsignedInt", "positiveInt", "markdown",
+                                  "url", "canonical", "uuid"]
+
+    def find_and_load_scruct_def_from_path(self, struct_def, path):
+
+        elements = struct_def['snapshot']["element"]
+
+        for elem in (elem for elem in elements if elem['id'] == path):
+
+            elem_type = elem['type']
 
-    def get_value_sets_for_code_filter(self, structDef):
+            for type_elem in (elem for elem in elem_type if re.search("Reference", elem['code'])):
+                struct_def_ref = type_elem['targetProfile'][0]
```

Comment:
Is always selecting the element at index 0 here safe? For instance, in the MII CDS Diagnose profile the subject element can reference both a Patient resource and a Group resource.

Reply:
@paulolaup  - no - this can be changed to filter for the mii reference - see comment below